### ISSUE TITLE
G28 w/ bed leveling probe only homing in corner.

### ISSUE THREAD

**Issue by iman7** [MarlinFirmware/Marlin#1542]
Makerfarm Prusa i3 
RAMPS 1.4
ReprapDiscount LCD controller
Marlin 1.0.2

When I finish a print from the SD card homing breaks in the sense that the LCD will display the right XY coordinates for the center of the bed but the hotend will stay at XY home and deploy the probe for z homing.

If I press the reset button on RAMPS soon after the print finishes it seems to fix the issue. However If I don't press it and leave it be for a while the method becomes ineffective and I have to plug into repetier-host  and home there which for some reason fixes it. 

Im assuming something weird is going on with Marlin because I've double checked the connections on the LCD controller and the motors and they both seem fine.

**Comment by thinkyhead** [comment]
@iman7 Homing with the Z probe is _relatively_ new code, so I wouldn't be surprised if it still has some kinks. So your i3 has no Z endstop, but does have a Z probe, or does it have both?

And are you using the latest (Development branch) version of Marlin, or the _tagged_ 1.0.2 version?

**Comment by iman7** [comment]
@thinkyhead I don't have a Z endstop, used it for the probe. As for the Marlin version. I downloaded the development branch as a zip file in the Feb 9, sorry Im not sure what you mean by the tagged 1.0.2 version. 

EDIT: Yes it was from the development branch from around Feb 9, I just realized there's a tags tab when you click on the branch button. Sorry rather new to git.

**Comment by thinkyhead** [comment]
@iman7 Welcome to the Github labyrinth... everything just 3 clicks away!

My best suggestion is to try the latest Development version, which has definitely had some bug-fixes since Feb 9, and see if it fixes the problem. If it doesn't, then we are at a good point to start investigating. We definitely want to fix all the bugs we can right now, looking forward to a 1.0.3 release.

**Comment by boelle** [comment]
but since you have Ramps why not have a Z endstop and probe? endstop to Z min and probe to Z max ?

**Comment by iman7** [comment]
@boelle I don't have access to those black connectors used on endstops locally. I don't feel like having to deal with customs for such a small part though. 

@thinkyhead 

Sorry for the delay. It's exam season where I am and had to put down tinkering for a while.
In that two week span the servo died on me and I went out to get a replacement, the replacement is much worse in the sense that it always homes only in the corner (Using the Feb 9 build). Unplugging the servo still causes it to home in the corner unlike previously where the head would move to the middle even without the servo. 

I also tried Marlin downloaded from git on Mar 11 The problem still persists. Unplugging the servo still causes the problem of homing in the corner and the lack of movement. 

Edit: To be clear the motors will hum as if they're moving but they wont move. They don't whine like when they're idling. 
Edit2: The motor issue was a wrong number placed in the steps per mm section of the firmware. My bad. 

Feels like its shoddy chinese servos but I don't know. 

[Link to Mar 11 config.h](http://pastebin.com/0W4gBVgM)
